tests/unit/browser/test_history.py

```diff
@@ -259,6 +259,41 @@ def test_clear_add_tab(self, web_history, mock_time):
         web_history.add_from_tab(QUrl(url), QUrl(url), 'title')
         assert list(web_history) == hist
 
+    @pytest.mark.parametrize(
+        'urls, expected', [
+            (['http://a', 'http://b', 'http://a'],
+             {'http://a': 2, 'http://b': 1}),
+            (['http://a/a a', 'http://b', 'http://a/a%20a'],
+             {'http://a/a a': 2, 'http://b': 1}),
+        ]
+    )
+    def test_visits(self, web_history, urls, expected):
+        for url in urls:
+            web_history.add_url(QUrl(url))
+        completion = {i.url: i.visits for i in web_history.completion}
+        assert completion == expected
+
+    @pytest.mark.parametrize(
+        'urls, expected', [
+            ([('http://a', 1)], [('http://a', lambda b: 1)]),
```

Comment:
You could parameterize the bonus as well, to test that the bonus config is respected.

Reply:
Good idea. Done in 6c973ac8022e9d5a30dd5a8de38de8328b845c39.